sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/ConsumerSpEL.java

```diff
@@ -57,4 +74,20 @@ public void evaluateAssign(Consumer consumer, Collection<TopicPartition> topicPa
     mapContext.setVariable("tp", topicPartitions);
     assignExpression.getValue(mapContext);
   }
+
+  public long getRecordTimestamp(ConsumerRecord<byte[], byte[]> rawRecord) {
+    long timestamp = -1L;
+    try {
+      //for Kafka 0.9, set to System.currentTimeMillis();
+      //for kafka 0.10, when NO_TIMESTAMP also set to System.currentTimeMillis();
+      if (!hasRecordTimestamp || (timestamp = (long) timestampMethod.invoke(rawRecord)) == -1L) {
```

Comment:
Unless I don't know of something specific that would cause a `-1` return value, and since there is a return value if the code gets here, I'd go with `< 0` instead of `== -1L`